packages/scenes-app/src/pages/Demos/scenes/behaviors.tsx

```diff
@@ -0,0 +1,85 @@
+import {
+  EmbeddedScene,
+  HiddenForTimeRangeBehavior,
+  HiddenWhenNoDataBehavior,
+  SceneControlsSpacer,
+  SceneFlexItem,
+  SceneFlexLayout,
+  SceneTimePicker,
+  SceneTimeRange,
+  SceneToolbarInput,
+  ShowBasedOnConditionBehavior,
+  VizPanel,
+} from '@grafana/scenes';
+import { SceneRadioToggle } from '../../../components/SceneRadioToggle';
+import { getQueryRunnerWithRandomWalkQuery, newTimeSeriesPanel } from '../utils';
+
+export function getBehaviorsDemo() {
+  const queryRunner = getQueryRunnerWithRandomWalkQuery({
+    seriesCount: 2,
+    alias: '__server_names',
+    scenarioId: 'random_walk',
+  });
+
+  const showHideToggle = new SceneRadioToggle({
+    options: [
+      { value: 'hidden', label: 'Show text panel' },
+      { value: 'visible', label: 'Hide text panel' },
+    ],
+    value: 'hidden',
+  });
+
+  return new EmbeddedScene({
+    $timeRange: new SceneTimeRange(),
+    controls: [
+      new SceneToolbarInput({
+        value: '2',
+        onChange: (newValue) => {
+          queryRunner.setState({
+            queries: [
+              {
+                ...queryRunner.state.queries[0],
+                seriesCount: newValue,
+              },
+            ],
+          });
+          queryRunner.runQueries();
+        },
+      }),
+      showHideToggle,
+      new SceneControlsSpacer(),
+      new SceneTimePicker({ isOnCanvas: true }),
+    ],
+    body: new SceneFlexLayout({
+      direction: 'column',
+      children: [
+        new SceneFlexItem({
+          $behaviors: [new ShowBasedOnConditionBehavior({ getCondition: getTextPanelToggle(showHideToggle) })],
+          body: new VizPanel({
+            pluginId: 'text',
+            options: { content: 'This panel can be hidden with a toggle!' },
+          }),
+        }),
+        new SceneFlexItem({
+          $behaviors: [new HiddenForTimeRangeBehavior({ greaterThan: 'now-2d' })],
+          body: newTimeSeriesPanel(
+            {
+              title: 'Hidden for time ranges > 2d',
+              $data: getQueryRunnerWithRandomWalkQuery(),
+            },
+            { fillOpacity: 20 }
+          ),
+        }),
+        new SceneFlexItem({
+          $behaviors: [new HiddenWhenNoDataBehavior()],
+          $data: queryRunner,
+          body: newTimeSeriesPanel({ title: 'Hidden when no time series' }),
+        }),
+      ],
+    }),
+  });
+}
+
+function getTextPanelToggle(toggle: SceneRadioToggle) {
+  return () => ({ references: [toggle], condition: () => toggle.state.value === 'visible' });
+}
```

Comment:
Another way to do this condition would be if references were only scene object keys, and the condition would get passed these objects. The benefit of this would be that the scene could be declared inline and the radio toggle variable would not have be declared before the rest of the scene 

```suggestion
function getTextPanelToggle() {
  return { references: ["toggle-key"], condition: (toggle: SceneRadioToggle) => toggle.state.value === 'visible' };
}
```